### PR TITLE
Remove CallerArgumentExpression feature flag

### PR DESCRIPTION
**Pull request by Youssef1313**
Relates to test plan https://github.com/dotnet/roslyn/issues/52745

### CODE REVIEW

File: src/Compilers/VisualBasic/Portable/Parser/ParserFeature.vb [hunk 1]
```diff
@@ -41,15 +41,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
         UnconstrainedTypeParameterInConditional
         CommentsAfterLineContinuation
         InitOnlySettersUsage
-        CallerArgumentExpression
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Parser/ParserFeature.vb:44]: >CallerArgumentExpression [](http://example.com/codeflow?start=8&length=24)

We are not doing language version checks on VB side?

Comment (Youssef1313) [reply src/Compilers/VisualBasic/Portable/Parser/ParserFeature.vb:44]: @AlekseyTs Not in C# as well.

**Review by 333fred — APPROVED** [review]
LGTM (commit 1)

**Comment by AlekseyTs** [comment]
>             Dim compilation = CreateCompilation(source, targetFramework:=TargetFramework.NetCoreApp, references:={Net451.MicrosoftVisualBasic}, options:=TestOptions.ReleaseExe, parseOptions:=TestOptions.RegularLatest.WithFeature("CallerArgumentExpression"))

It feels like all references to the feature should be removed.

---
Refers to: src/Compilers/VisualBasic/Test/Emit/Attributes/AttributeTests_CallerArgumentExpression.vb:33 in 7cd5ceb. [](commit_id = 7cd5cebb68c5f06fa5daa3077d921c89c9e1841b, deletion_comment = False)

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by 333fred** [comment]
Thanks @Youssef1313! 🎉

**Pull request merged**